pennylane/transforms/core/transform_dispatcher.py

```diff
@@ -205,7 +205,7 @@ def default_qnode_transform(self, qnode, targs, tkwargs):
             qnode._original_device.reset()
             qnode.device.reset()
```

Comment:
If I a not wrong you can remove the if statement and reset of device.

Reply:
I don't understand why these lines are there, so I don't feel comfortable touching them in this PR.

Reply:
We had some issues with copying previous generation devices when it was jitting because the state would be abstract and deepcopying would fail hence the reset. We can remove it in another PR if you prefer